dev/devicelab/lib/tasks/perf_tests.dart

```diff
@@ -1339,16 +1348,55 @@ class WebCompileTest {
     });
   }
 
-  /// Obtains the size and gzipped size of a file given by [fileName].
-  static Future<Map<String, int>> getSize(String fileName, {required String metric}) async {
+  /// Obtains the size and gzipped size of both [dartBundleFile] and [buildDir].
+  static Future<Map<String, int>> getSize({
+    /// Mapping of metric key name to file system path for directories to measure
+    Map<String, String> directories = const <String, String>{},
+    /// Mapping of metric key name to file system path for files to measure
+    Map<String, String> files = const <String, String>{},
+    required String metric,
+  }) async {
     final Map<String, int> sizeMetrics = <String, int>{};
 
-    final ProcessResult result = await Process.run('du', <String>['-k', fileName]);
-    sizeMetrics['${metric}_dart2js_size'] = _parseDu(result.stdout as String);
+    final Directory tempDir = Directory.systemTemp.createTempSync('perf_tests_gzips');
+    for (final MapEntry<String, String> entry in files.entries) {
+      final String key = entry.key;
+      final String filePath = entry.value;
+      ProcessResult result = await Process.run('du', <String>['-k', filePath]);
+      sizeMetrics['${metric}_${key}_size'] = _parseDu(result.stdout as String);
+
+      await Process.run('gzip',<String>['--keep', '-9', filePath]);
+      // gzip does not provide a CLI option to specify an output file, so
+      // instead just move the output file to the temp dir
```

Comment:
You may be able to gzip to a specific location with `-c` and a pipe, something like this:

```
gzip --keep -9 -c filePath > compressedFilePath
```

That way there's no need to mv later (or to concatenate the path of the compressedFilePath for the `du` call)

Reply:
I don't think you can pipe in a `Process.run()` call, as it's not using a shell

Reply:
Ooooh, bummer! (however if you can use the stdout stream from Process.run, you may be able to directly write it to the destination!) anyway: unimportant. Thanks for looking into the piping!